x-pack/legacy/plugins/index_management/public/components/template_form/template_form_schemas.tsx

```diff
@@ -0,0 +1,132 @@
+  fieldValidators,
+} from '../../../../../../../src/plugins/es_ui_shared/static/forms/helpers';
+
+import {
+  INVALID_INDEX_PATTERN_CHARS,
+  INVALID_TEMPLATE_NAME_CHARS,
+} from '../../../common/constants';
+
+const { emptyField, containsCharsField, startsWithField, indexPatternField } = fieldValidators;
+const { toInt } = fieldFormatters;
+const indexPatternInvalidCharacters = INVALID_INDEX_PATTERN_CHARS.join(' ');
+
+export const schemas: Record<string, FormSchema> = {
+  logistics: {
+    name: {
+      type: FIELD_TYPES.TEXT,
+      label: i18n.translate('xpack.idxMgmt.templateForm.stepLogistics.fieldNameLabel', {
+        defaultMessage: 'Name',
+      }),
+      validations: [
+        {
+          validator: emptyField(
+            i18n.translate('xpack.idxMgmt.templateValidation.templateNameRequiredError', {
+              defaultMessage: 'A template name is required.',
+            })
+          ),
+        },
+        {
+          validator: containsCharsField({
+            chars: ' ',
+            message: i18n.translate('xpack.idxMgmt.templateValidation.templateNameSpacesError', {
+              defaultMessage: 'Spaces are not allowed in a template name.',
+            }),
+          }),
+        },
+        {
+          validator: startsWithField({
+            char: '_',
+            message: i18n.translate(
+              'xpack.idxMgmt.templateValidation.templateNameUnderscoreError',
+              {
+                defaultMessage: 'A template name must not start with an underscore.',
+              }
+            ),
+          }),
+        },
+        {
+          validator: startsWithField({
+            char: '.',
+            message: i18n.translate('xpack.idxMgmt.templateValidation.templateNamePeriodError', {
+              defaultMessage: 'A template name must not start with a period.',
+            }),
+          }),
+        },
+        {
+          validator: containsCharsField({
+            chars: INVALID_TEMPLATE_NAME_CHARS,
+            message: ({ charsFound }) =>
+              i18n.translate(
+                'xpack.idxMgmt.templateValidation.templateNameInvalidaCharacterError',
+                {
+                  defaultMessage: 'A template name must not contain the character "{invalidChar}"',
+                  values: { invalidChar: charsFound[0] },
+                }
+              ),
+          }),
+        },
+      ],
+    },
+    indexPatterns: {
+      type: FIELD_TYPES.COMBO_BOX,
+      defaultValue: [],
+      label: i18n.translate('xpack.idxMgmt.templateForm.stepLogistics.fieldIndexPatternsLabel', {
+        defaultMessage: 'Index patterns',
+      }),
+      helpText: (
+        <FormattedMessage
+          id="xpack.idxMgmt.templateForm.stepLogistics.fieldIndexPatternsHelpText"
+          defaultMessage="Spaces and the characters {invalidCharactersList} are not allowed."
+          values={{
+            invalidCharactersList: <strong>{indexPatternInvalidCharacters}</strong>,
```

Comment:
I think this should probably be derived from the same characters defined in the `indexPatternField` validator. WDYT?

Reply:
Yes, it's a good point. Although the `indexPatternField` imports the chars from `'ui/index_patterns'` which at some point we will need to remove the dependency from. I guess that's why in the first place you manually added the constant in the app?

I am looking at the index patterns folder (now moved to core_plugins/data/public/index_patterns) and there is quite some mess. 😊 

[EDIT] I tried to import from 'ui/index_patterns' and also relative '../../src/...' to import the constant but was not allowed. Will leave this as is for now.